exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/ConvertCountToDirectScanRule.java

```diff
@@ -0,0 +1,296 @@
+   * @param agg aggregate relational expression
+   * @param scan scan relational expression
+   * @param project project relational expression
+   * @return result map where key is count column name, value is count value
+   */
+  private Map<String, Long> collectCounts(PlannerSettings settings, Metadata_V4.MetadataSummary metadataSummary,
+                                          Aggregate agg, TableScan scan, Project project) {
+    final Set<String> implicitColumnsNames = ColumnExplorer.initImplicitFileColumns(settings.getOptions()).keySet();
+    final long totalRecordCount = metadataSummary.getTotalRowCount();
+    final LinkedHashMap<String, Long> result = new LinkedHashMap<>();
+
+    for (int i = 0; i < agg.getAggCallList().size(); i++) {
+      AggregateCall aggCall = agg.getAggCallList().get(i);
+      long cnt;
+
+      // rule can be applied only for count function, return empty counts
+      if (!"count".equalsIgnoreCase(aggCall.getAggregation().getName()) ) {
+        return ImmutableMap.of();
+      }
+
+      if (CountToDirectScanUtils.containsStarOrNotNullInput(aggCall, agg)) {
+        cnt = totalRecordCount;
+
+      } else if (aggCall.getArgList().size() == 1) {
+        // count(columnName) ==> Agg ( Scan )) ==> columnValueCount
+        int index = aggCall.getArgList().get(0);
+
+        if (project != null) {
+          // project in the middle of Agg and Scan : Only when input of AggCall is a RexInputRef in Project, we find the index of Scan's field.
+          // For instance,
+          // Agg - count($0)
+          //  \
+          //  Proj - Exp={$1}
+          //    \
+          //   Scan (col1, col2).
+          // return count of "col2" in Scan's metadata, if found.
+          if (!(project.getProjects().get(index) instanceof RexInputRef)) {
+            return ImmutableMap.of(); // do not apply for all other cases.
+          }
+
+          index = ((RexInputRef) project.getProjects().get(index)).getIndex();
+        }
+
+        String columnName = scan.getRowType().getFieldNames().get(index).toLowerCase();
+
+        // for implicit column count will be the same as total record count
+        if (implicitColumnsNames.contains(columnName)) {
+          cnt = totalRecordCount;
+        } else {
+          SchemaPath simplePath = SchemaPath.getSimplePath(columnName);
+
+          if (ColumnExplorer.isPartitionColumn(settings.getOptions(), simplePath)) {
+            return ImmutableMap.of();
+          }
+
+          Metadata_V4.ColumnTypeMetadata_v4 columnMetadata = metadataSummary.getColumnTypeInfo(new Metadata_V4.ColumnTypeMetadata_v4.Key(simplePath));
+
+         if (columnMetadata == null) {
+            // if column stats is not available don't apply this rule, return empty counts
+            return ImmutableMap.of();
+          } else {
+           // count of a nullable column = (total row count - column's null count)
+           cnt = totalRecordCount - columnMetadata.totalNullCount;
```

Comment:
It is mostly a question to the previous commit. For V3 and older versions were possible cases when statistics were unavailable, but its `ColumnTypeMetadata` was still present. For such cases, `nullsCout` was set to `-1` (`GroupScan.NO_COLUMN_STATS`).
Should this check be added here, or v4 does not store `ColumnTypeMetadata` for such columns?

Reply:
In the PR for DRILL-7063, I do see the initialization of `nullsCount` to `GroupScan.NO_COLUMN_STATS` for `ColumnTypeMetadata_V4`: 

`if (parquetTableMetadata instanceof Metadata_V4.ParquetTableMetadata_v4) {
	      for (Metadata_V4.ColumnTypeMetadata_v4 columnTypeMetadata :
	          ((Metadata_V4.ParquetTableMetadata_v4) parquetTableMetadata).getColumnTypeInfoMap().values()) {
	        SchemaPath schemaPath = SchemaPath.getCompoundPath(columnTypeMetadata.name);
	        if (!schemaPaths.contains(schemaPath)) {
	          Map<String, Object> statistics = new HashMap<>();
	          statistics.put(ColumnStatisticsKind.NULLS_COUNT.getName(), GroupScan.NO_COLUMN_STATS);`

I would like @dvjyothsna to confirm if this is happening for all columns (regardless of interesting/non-interesting).

Reply:
@vvysotskyi Just to clarify your question is what will be the nullCount in ColumnTypeMetadata. The null count is set to -1 as we did for V3. If the statistics are not present, nullCount will be set to -1 in [here](https://github.com/apache/drill/blob/bd5c530262dfe05ce21e1a7d2a5822553fe82148/exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/metadata/Metadata.java#L559). This is for all the columns i.e., both the interesting and non-interesting columns.

Reply:
Thanks for the clarification, in this case, should be returned empty map, since rule cannot be applied if `columnMetadata.totalNullCount == GroupScan.NO_COLUMN_STATS`

Reply:
@amansinha100, could you please address this comment also?

Reply:
Sorry, I somehow thought this comment was for @dvjyothsna .  I will add this check and do an early return if columnMetadata is non-null but the totalNullCount stats is not available.